website/pages/docs/k8s/connect/health.mdx

```diff
@@ -0,0 +1,47 @@
+---
+layout: docs
+page_title: Kubernetes Health Checks
+sidebar_title: Health Checks
+description: Configuring Kubernetes Health Checks
+---
+
+# Kubernetes Health Checks in Consul on Kubernetes
+
+-> 0.25+ This feature is available in consul-helm versions 0.25 and higher
+
+~> This topic requires familiarity with [Kubernetes Health Checks](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/).
+
+This page describes how to enable Kubernetes Health checks to extend Consul for service mesh uses cases.
+
+When deploying your helm chart, you must provide helm with a custom yaml file that contains your environment configuration.
+Enabling health checks is done via the `healthChecks` stanza under `connectInject`.
+
+When `enabled: true` is set, a [TTL health check](https://www.consul.io/docs/discovery/checks) will be registered within
+Consul for each Kubernetes pod that is connect-injected. The Consul health check's state will reflect the pod's readiness
+status, which is the combination of all probes registered with the pod. Kubernetes Readiness probes are the primary use
+case as they can change state without restarting the containers in the pod.
+
+When readiness probes are set for a pod, the status of the pod will be reflected within Consul, and will cause Consul to redirect service
+mesh traffic to the pod based on the pod's health. If the pod were to have failing health checks, Consul will no longer use
+the service instance associated with the pod for service mesh traffic. When the pod passes its health checks, Consul will
+then use the respective service instance for service mesh traffic.
+
+In the case where no health checks are assigned to a pod, the default behavior is that the Consul health check will always
+be marked `passing`.
```

Comment:
🤔 since we're syncing the ready status, wouldn't we always be able to do that even when a pod doesn't have readiness probes defined?

Reply:
I believe that is what I'm trying to say : In the absence of user defined readiness probes a consul health check will still be created and will always be passing.

Reply:
Oh, I thought we will still update it when the pod changes readiness status. I think it's possible for readiness status to be changed even without readiness probes defined, e.g when a container crashes. In that case, we will still update the TTL health check, or am I misunderstanding it?

Reply:
Ah you are spot on, that is a very good point, let me try to re-word it :-)